xcode_parser.rb

```diff
@@ -19,19 +19,35 @@ def execute_cmd(cmd)
     output
   end
 
+  @@xcode_version = nil
+  def get_xcode_version
+    @@xcode_version ||= begin
+      output = execute_cmd("xcodebuild -version")
+      output.match(/Xcode (\d+(\.\d+)?)/)[1].to_f
+    end
+  end
+
+  def get_xcrun_command(cmd_func, id: nil, filename: nil, output_path: nil)
+    cmd = "xcrun xcresulttool #{cmd_func}"
+    cmd += " object --legacy"  if get_xcode_version >= 16.0
+    cmd += " --format json --path #{@test_path}" if cmd_func == "get"
+    cmd += " --id #{id}" if cmd_func == "get" && id
+    cmd += " --path #{@test_path} --id #{id} --output-path '#{output_path}' --type file" if output_path
+    cmd
+  end
```

Comment:
@boztopuz There is code duplication in the following parameters:
- `--path #{@test_path}` 
- `--id #{id}`

Is there anything preventing you from using it as follows?
```suggestion
  def get_xcrun_command(cmd_func, id: nil, filename: nil, output_path: nil)
    cmd = "xcrun xcresulttool #{cmd_func}"
    cmd += " object --legacy"  if get_xcode_version >= 16.0
    cmd += " --format json" if cmd_func == "get"
    cmd += " --path #{@test_path}"
    cmd += " --id #{id}" if id
    cmd += " --output-path '#{output_path}' --type file" if output_path
    cmd
  end
```

When I checked, the above code does not break the args order and does not contain repetition (Complies with the DRY principle).